Samples/iOS-Swift/iOS-Swift-UITests/UserFeedbackUITests.swift

```diff
@@ -0,0 +1,178 @@
+        // the placeholder text is returned for XCUIElement.value
+        XCTAssertEqual(try XCTUnwrap(nameField.value as? String), "Your Name")
+        XCTAssertEqual(try XCTUnwrap(emailField.value as? String), "your.email@example.org")
+        
+        // the UITextView doesn't hav a placeholder, it's a label on top of it. so it is actually empty
+        XCTAssertEqual(try XCTUnwrap(messageTextView.value as? String), "")
+    }
+    
+    func testCancelFromFormBySwipeDown() {
+        widgetButton.tap()
+        
+        // fill out the fields; we'll assert later that the entered data does not reappear on subsequent displays
+        nameField.tap()
+        nameField.typeText("Andrew")
+        
+        emailField.tap()
+        emailField.typeText("andrew.mcknight@sentry.io")
+        
+        messageTextView.tap()
+        messageTextView.typeText("UITest user feedback")
+        
+        // the cancel gesture
+        app.swipeDown(velocity: .fast)
+        app.swipeDown(velocity: .fast)
+        
+        // the swipe dismiss animation takes an extra moment, so we need to wait for the widget to be visible again
+        XCTAssert(widgetButton.waitForExistence(timeout: 1))
+        
+        // displaying the form again ensures the widget button still works afterwards; also assert that the fields are in their default state to ensure the entered data is not persisted between displays
+        
+        widgetButton.tap()
+        
+        // the placeholder text is returned for XCUIElement.value
+        XCTAssertEqual(try XCTUnwrap(nameField.value as? String), "Your Name")
+        XCTAssertEqual(try XCTUnwrap(emailField.value as? String), "your.email@example.org")
+        
+        // the UITextView doesn't hav a placeholder, it's a label on top of it. so it is actually empty
+        XCTAssertEqual(try XCTUnwrap(messageTextView.value as? String), "")
+    }
+    
+    func testAddingAndRemovingScreenshots() {
+        widgetButton.tap()
+        addScreenshotButton.tap()
+        XCTAssert(removeScreenshotButton.isHittable)
+        XCTAssertFalse(addScreenshotButton.isHittable)
+        removeScreenshotButton.tap()
+        XCTAssert(addScreenshotButton.isHittable)
+        XCTAssertFalse(removeScreenshotButton.isHittable)
+    }
+    
+    // MARK: Private
+    
+    var widgetButton: XCUIElement {
+        app.otherElements["io.sentry.feedback.widget"]
+    }
+    
+    var nameField: XCUIElement {
+        app.textFields["io.sentry.feedback.form.name"]
+    }
+    
+    var emailField: XCUIElement {
+        app.textFields["io.sentry.feedback.form.email"]
+    }
+    
+    var messageTextView: XCUIElement {
+        app.textViews["io.sentry.feedback.form.message"]
+    }
+    
+    var addScreenshotButton: XCUIElement {
+        app.buttons["io.sentry.feedback.form.add-screenshot"]
+    }
+    
+    var removeScreenshotButton: XCUIElement {
+        app.buttons["io.sentry.feedback.form.remove-screenshot"]
+    }
+}
```

Comment:
💯 I like that approach. It improves the readibilty and maintainability of the tests.